File: src/algorithms/MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int> &num) {
        int ret = 0, count = 0;
        map<int, int> map;
        for (int i = 0; i < num.size(); i++) {
        	map[num[i]]++;
        }
        for (int i = 0; i < num.size(); i++) {
        	int tt = map[num[i]];
        	if (count < tt) {
        	    count = tt;
        	    ret = num[i];
        	}
        }
        return ret;
    }
};

File: Algorithms/python/169-MajorityElement/MajorityElement.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums_count = dict()

        for v in nums:
            if v not in nums_count:
                c = 1
            else:
                c = nums_count[v] + 1

            if c > len(nums) / 2:
                return v
            nums_count[v] = c
